app/src/org/commcare/utils/StringUtils.java

```diff
@@ -71,4 +66,19 @@ public static String convertPairToJsonString(Pair<? extends Serializable, ? exte
             return null;
         }
     }
+
+    public static Pair<? extends Serializable, ? extends Serializable> convertJsonStringToPair(
+            String stringInJsonFormat) {
+        Gson gson = new Gson();
+        if (stringInJsonFormat == null) {
+            return null;
+        }
+
+        try{
+            return gson.fromJson(stringInJsonFormat, Pair.class);
+        } catch(JsonIOException e){
+            // default to null
+            return null;
```

Comment:
we should log all catched exceptions here, similar comment for pair to JSON conversion method.